indexer/bin/indexer.js

```diff
@@ -50,148 +55,167 @@ export default {
-              try {
-                return await pdpVerifier.getRootCid(
-                  setId,
-                  BigInt(rootId),
-                  payload.block_number,
-                )
-              } catch (/** @type {any} */ err) {
-                console.error(
-                  `RootsAdded: Cannot resolve root CID for setId=${setId} rootId=${rootId}: ${err?.stack ?? err}`,
-                )
-                throw err
-              }
-            }),
-          )
+      const pieceIds = payload.piece_ids.split(',')
+      /** @type {string[]} */
+      const pieceCids = payload.piece_cids.split(',')
 
       console.log(
-        `New roots (root_ids=[${rootIds.join(', ')}], root_cids=[${rootCids.join(', ')}], set_id=${payload.set_id})`,
+        `New pieces (piece_ids=[${pieceIds.join(', ')}], piece_cids=[${pieceCids.join(', ')}], data_set_id=${payload.set_id})`,
       )
 
-      await insertProofSetRoots(env, payload.set_id, rootIds, rootCids)
+      await insertDataSetPieces(env, payload.set_id, pieceIds, pieceCids)
 
       return new Response('OK', { status: 200 })
-    } else if (pathname === '/roots-removed') {
+    } else if (pathname === '/pdp-verifier/pieces-removed') {
       if (
         !(
           typeof payload.set_id === 'number' ||
           typeof payload.set_id === 'string'
         ) ||
-        !payload.root_ids ||
-        typeof payload.root_ids !== 'string'
+        !payload.piece_ids ||
+        typeof payload.piece_ids !== 'string'
       ) {
-        console.error('RootsRemoved: Invalid payload', payload)
+        console.error('PDPVerifier.PiecesRemoved: Invalid payload', payload)
         return new Response('Bad Request', { status: 400 })
       }
 
       /** @type {string[]} */
-      const rootIds = payload.root_ids.split(',')
+      const pieceIds = payload.piece_ids.split(',')
 
       console.log(
-        `Removing roots (root_ids=[${rootIds.join(', ')}], set_id=${payload.set_id})`,
+        `Removing pieces (piece_ids=[${pieceIds.join(', ')}], data_set_id=${payload.set_id})`,
       )
 
-      await removeProofSetRoots(env, payload.set_id, rootIds)
+      await removeDataSetPieces(env, payload.set_id, pieceIds)
       return new Response('OK', { status: 200 })
-    } else if (pathname === '/proof-set-rail-created') {
+    } else if (pathname === '/filecoin-warm-storage-service/data-set-created') {
       if (
-        !payload.proof_set_id ||
+        !payload.data_set_id ||
         !(
-          typeof payload.proof_set_id === 'number' ||
-          typeof payload.proof_set_id === 'string'
-        ) ||
-        !payload.rail_id ||
-        !(
-          typeof payload.rail_id === 'number' ||
-          typeof payload.rail_id === 'string'
+          typeof payload.data_set_id === 'number' ||
+          typeof payload.data_set_id === 'string'
         ) ||
         !payload.payer ||
-        !payload.payee
+        !payload.payee ||
+        typeof payload.with_cdn !== 'boolean'
```

Comment:
Looks like the `with_cdn` flag has been moved to metadata keys and values inside the `DataSetCreated` event

https://github.com/FilOzone/filecoin-services/blob/f9a8ec78c8d75ecc0c020482cd738d8fefedac30/service_contracts/src/FilecoinWarmStorageService.sol#L47-L56

Reply:
I will address this in a follow up PR, I've discussed with @bajtos that we will land this first and then create follow up PRs for any more recent changes. For context, when I opened this PR, the metadata keys PR hadn't landed yet